debugtools/DDR_VM/src/com/ibm/j9ddr/vm29/tools/ddrinteractive/commands/ShrCCommand.java

```diff
@@ -382,7 +386,28 @@ private void dbgShrcCacheName(PrintStream out, J9SharedClassConfigPointer shared
 			}
 		}
 	}
-	
+
+	private int dbgShrcCacheTopLayer(PrintStream out, J9SharedClassConfigPointer sharedClassConfig) throws CorruptDataException {
+		SH_OSCachePointer osCache = getOSCache(out, sharedClassConfig);
+		int topLayer = 0;
+		if (osCache.notNull()) {
+			U8Pointer cacheNamePointer = osCache._cacheNameWithVGen();
+			if (cacheNamePointer.notNull()) {
+				String cacheNameString = cacheNamePointer.getCStringAtOffset(0);
+				Pattern p = Pattern.compile(".*L\\d\\d$");
```

Comment:
Parentheses (`".*L(\\d\\d)$"`) will allow extraction of the layer directly as `m.group(1)` below.

Reply:
Ah, I see. It is fixed now.